### PR TITLE
JavaScript DataElement expansion

### PR DESCRIPTION
**Pull request by ljades**
NOTE: This is intended to merge into patient_fix, as it is based on that branch and I need both in in order for the functionality in cql_calculator to work.

Restructured and refactored JavaScript Data Element object to better function as the schema extension needed for other elements. Data Elements and Intervals can now also have their properties invoked by camel case alias.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR:
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] If there were any JavaScript changes, this PR has updated the `dist` directory using `npm run dist`
- [x] If applicable, the library version number in `package.json` and `cqm-models.gemspec` has been updated

**Reviewer 1:**

Name: @holmesie
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name: @okeefm
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 3:**

Name: @jbradl11 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: app/assets/javascripts/basetypes/DataElement.js [hunk 1]
```diff
@@ -1,36 +1,69 @@
...
+
+    if (schema) {
+      obj = schema.applyGetters(obj, this);
+    }
+
+    return obj;
+  };
+
+  // Returns all of the codes on this data element.
+  extended.methods.getCode = function getCode() {
```
Comment (jbradl11) [inline app/assets/javascripts/basetypes/DataElement.js]: This function and the function below it, 'getCodes()' have duplicate comments. Please elaborate on the differences between the two functions.

File: app/assets/javascripts/basetypes/DataElement.js [hunk 2]
```diff
@@ -1,36 +1,69 @@
...
+
+  if (add) {
+    extended.add(add);
+  }
+
+  extended.methods.get = function get(path, type) {
+    let adhoc;
+    const pathSnake = changeCase.snakeCase(path);
+    if (type) {
+      adhoc = Schema.interpretAsType(path, type, this.schema.options) || Schema.interpretAsType(pathSnake, type, this.schema.options);
```
Comment (jbradl11) [inline app/assets/javascripts/basetypes/DataElement.js]: Please add some comments for this get function. Seems a like a big change from the previous version of "get()"

File: app/assets/javascripts/basetypes/DataElement.js [hunk 3]
```diff
@@ -1,36 +1,74 @@
...
+  const extended = new Schema({
+    data_element_codes: { type: [Code] },
+    description: { type: String },
+  }, options);
+
+  if (add) {
+    extended.add(add);
+  }
+
+  // TODO: REMOVE THIS. No longer needed after the change to camelCase
```
Comment (jbradl11) [inline app/assets/javascripts/basetypes/DataElement.js]: Is something preventing you from removing this now?

Comment (ljades) [reply app/assets/javascripts/basetypes/DataElement.js]: Yes. Until another PR is in that changes every patient and data element field to camel case, this conversion needs to happen.
